### DISCUSSION TITLE
Open npz data

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [saulpw/visidata#1440] [Q&A]
Hello every one 
I recently found a way to use Visidata (even in Google colab).
It's with the help of `xterm`

I want to open a dataset with the format of `.npz`
It is a numpy array for university project.

it seems that visidata does not support it.
Any tips or helps will be appreciated.

**Comment by anjakefala** [comment]
What problems are you seeing?

`.npz` has been supported since v2.0! You need tg have `numpy` in the environment, but otherwise it should work automagically.

**Reply by Shahin-rmz** [reply]
Thanks 
All I see is this empty table.
Other datasets with csv format can be open without problem.

![image](https://user-images.githubusercontent.com/32805117/180595511-c672822b-c28f-4189-be25-dc84e663c689.png)

**Reply by Shahin-rmz** [reply]
PS. Numpy has been already imported

**Reply by saulpw** [reply]
You'd have to share the actual .npz, @Shahin-rmz.  Without that, there's nothing to go on.

**Reply by Shahin-rmz** [reply]
> You'd have to share the actual .npz, @Shahin-rmz. Without that, there's nothing to go on.

[Here you go](https://we.tl/t-Udeg7Dby5t).
it's a neuroscience public dataset of neural activity of some rats.
It has 3 parts and it is first part of that.
I'm not sure if I should merge and send 3 parts together or this part alone works.
Best

**Reply by Shahin-rmz** [reply]
Did you have any chance to take a look at the data?

**Reply by anjakefala** [reply]
@Shahin-rmz From what I can tell, @saulpw has pushed a fix on the `develop` branch for the `.npz` loader.

https://github.com/saulpw/visidata/pull/1493/commits/97dcff126ebb00e3088ffdb260b277275ed8bbf4

There will be a VisiData release soon with this fix. If you want to check it out before the release, please upgrade your VisiData with:

`pip3 install --upgrade git+https://github.com/saulpw/visidata.git`.

**Reply by saulpw** [reply]
Hi @Shahin-rmz, I can't download the data anymore.  Can you please upload a small .npy or .npz that exhibits this problem, instead of using a link to separate sharing site that expires?
